audio_clip_encoder.py

```diff
@@ -30,36 +29,35 @@ def __init__(self,
         self.aclp = AudioCLIP(pretrained=model_path)
         self.audio_transforms = ToTensor1D()
         self.default_traversal_paths = default_traversal_paths
+        self.default_batch_size = default_batch_size
 
     @requests
     def encode(self, docs: Optional[DocumentArray], parameters: dict, *args, **kwargs) -> Any:
 
         if docs:
             cleaned_document_array = self._get_input_data(docs, parameters)
-            self._create_embeddings(cleaned_document_array)
+            document_batches_generator = get_docs_batch_generator(
+                cleaned_document_array,
+                traversal_path=parameters.get('traversal_paths', self.default_traversal_paths),
+                batch_size=parameters.get('batch_size', self.default_batch_size),
+                needs_attr='blob',
+            )
+            self._create_embeddings(document_batches_generator)
 
     def _get_input_data(self, docs: DocumentArray, parameters: dict):
         """Create a filtered set of Documents to iterate over."""
-
-        traversal_paths = parameters.get('traversal_paths', self.default_traversal_paths)
-
-        # traverse thought all documents which have to be processed
-        flat_docs = docs.traverse_flat(traversal_paths)
-
         # filter out documents without images
-        filtered_docs = DocumentArray([doc for doc in flat_docs if doc.blob is not None])
+        filtered_docs = DocumentArray([doc for doc in docs if doc.blob is not None])
 
         return filtered_docs
 
-    def _create_embeddings(self, filtered_docs: Iterable):
-        """Update the documents with the embeddings generated by VGGISH"""
-
-        for d in filtered_docs:
-            # Vggish broadcasts across different length audios, not batches
-            audio = [d.blob, deepcopy(d.blob)]
-            audio = torch.stack([self.audio_transforms(track.reshape(1, -1)) for track in audio])
-            ((audio_features, _, _), _), _ = self.aclp(audio=audio)
-            audio_features = audio_features[0]
-            audio_features = audio_features / torch.linalg.norm(audio_features, dim=-1, keepdim=True)
+    def _create_embeddings(self, document_batches_generator: Iterable):
+        """Update the documents with the embeddings generated by AudioCLIP"""
 
-            d.embedding = audio_features.numpy()
+        for document_batch in document_batches_generator:
+            audio = torch.stack([self.audio_transforms(d.blob.reshape(1, -1)) for d in document_batch])
+            ((embedding_batch, _, _), _), _ = self.aclp(audio=audio)
+            embedding_batch = embedding_batch / torch.linalg.norm(embedding_batch, dim=-1, keepdim=True)
```

Comment:
why do we do this?

Reply:
these are preprocessing and postprocessing steps performed in the AudioCLIP's demo in order to get embeddings useful for query: https://github.com/AndreyGuzhov/AudioCLIP/blob/master/demo/AudioCLIP.ipynb

Reply:
this is not used in the `CLIPTextEncoder` so it cannot be here for now.

See that this is postprocess needed to have a classification layer, but to work on the embedding space it can be good.

For now, we need to remove it and we can discuss if we need to add this `normalization` step in the other `CLIP` executors